androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/playerOverlay/Settings.kt

```diff
@@ -65,15 +77,71 @@ fun Settings(
     }
 }
 
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TrackSelector(
+    label: String,
+    selected: String,
+    tracks: ArrayList<Pair<String, TrackSelectionParameters.Builder>>,
+    onTrackSelected: (Pair<String, TrackSelectionParameters.Builder>) -> Unit,
+    modifier: Modifier = Modifier,
+    icon: @Composable () -> Unit = {}
+) {
+    if (tracks.size > 2) {
+        var isSelectorOpen by rememberSaveable { mutableStateOf(false) }
+        val sheetState = rememberModalBottomSheetState()
+        val scope = rememberCoroutineScope()
+
+        ListItem(
+            headlineContent = { Text(label) },
+            leadingContent = icon,
+            trailingContent = { Text(selected) },
+            modifier = modifier
+                .clickable(onClick = { isSelectorOpen = true })
+                .testTag("${label}SettingsButton")
+        )
+
+        if (isSelectorOpen) {
+            ModalBottomSheet(
+                onDismissRequest = { isSelectorOpen = false },
+                sheetState = sheetState,
+                modifier = Modifier.testTag("${label}SettingsSelector")
+            ) {
+                tracks.forEach { item ->
```

Comment:
you will need to create a lazy list here, because the number of item can be greather than the vertical space available